ee/WFLY-1160-Provide_ability_to_configure_a_global_directory_which_puts_the_contents_in_the_classpath.adoc

```diff
@@ -0,0 +1,130 @@
+
+The simplest way to accomplished this is to use the EE subsystem to configure a path to the designated directory that will contain the libraries we want to be shared across all deployed applications. A new resource named `global-directory` will be added to the EE subsystem. This resource will use two attributes path and relative-to to configure a global directory:
+
+  [standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:add(path=lib, relative-to=jboss.home.dir)
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee/global-directory=my-common-libs:add(path=lib, relative-to=jboss.server.data.dir)
+
+The following attributes will be available on the global-directory resource:
+
+* path: The path of the directory to scan. (Mandatory)
+* relative-to: The name of another previously named path, or of one of the standard paths provided by the system. (Optional)
+
+When a new `global-directory` is created, the server will create a module that will have as resources the directory created using 'path' and 'relative-to' attributes and one resource-root for each jar file included in this directory. Subdirectories will not be scanned.
+
+The resources of this module will be created iterating over all jar files in alphabetical order and size ascending. The resource root order declaration will govern the class loading order for the classes available from the resources of this module.
+
+e.g. Suppose that our global directory path is /lib and it contains two jar files, lib-1.jar and lib-2.jar. The module generated internally by the global-directory operation will be equivalent to declare a module.xml with the following resources:
+
+  <resources>
+    <resource-root path="/lib"/>
+    <resource-root path="/lib/lib-1.jar"/>
+    <resource-root path="/lib/lib-2.jar"/>
+  </resources>
+
+The module created from the shared directory will be loaded as soon as the first application is deployed in the server. That means, if the server is started/restarted and there are no applications deployed, then the shared directory is neither scanned nor the module loaded.
+
+In case of domain mode or distributed environments, it is the user responsibility to make the content of the configured global directory consistent across all the server instances, as well as distribute the jar files that they contain.
+
+Removal of a global-directory could be done using the following operation:
+
+  [standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:remove()
+
+  [domain@localhost:9990 /] /profile=default/subsystem=ee/global-directory=my-common-libs:remove()
+
+As any other resources, users can read the current configuration of a global-directory using resource-resource operation:
+
+[standalone@localhost:9990 /] /subsystem=ee/global-directory=my-common-libs:read-resource
+{
+    "outcome" => "success",
+    "result" => {
```

Comment:
Can be here more information about the shared library?
E.q.:
- If is the shared library used/loaded
- If loading is succeeded of thrown Exception?

Reply:
Not really, this operation only returns the values of this resource. 

`If is the shared library used/loaded`

As soon as there is an application deployed, the library is loaded

`If loading is succeeded of thrown Exception?`

If the library cannot be loaded, then the application deployment cannot be done, and the deployment will fail.

Libraries in the shared directory will be loaded when the deployment is done. Usually, the users can define one global directory, copy the jar files in it, then restart the server to apply the changes to the deployed application. If a library cannot be load, deployments will fail.

Reply:
Ok